src/js/features/data/makeGetDataRequest.js

```diff
@@ -28,14 +39,14 @@ export const makeGetDataRequest = createAsyncThunk(
       // converting fields to usable form
       let fields = {
         ...queryParameterStack,
-        ...queryParameterStack.extra_properties,
+        ...(queryParameterStack.extra_properties ?? {}),
       };
-      delete fields.extra_properties;
+      if (fields.extra_properties) delete fields.extra_properties;
```

Comment:
This _works_, but is the goal here just to remove the extra nesting around what's inside extra_properties? This can be done with destructuring: 

```
const {extra_properties, ...everything_else} = queryParameterStack
let fields = {...everything_else, ...extra_properties}
```

which I find easier to understand, and requires no extra checks to see if an extra_properties field is present.

Reply:
This looks nice!
I've never realized that `undefined` could be spread. Thanks!

Reply:
haha! I didn't notice it either until today.